storage/memory/peer_store.go

```diff
@@ -92,6 +92,14 @@ func (s *peerStore) shardIndex(infoHash bittorrent.InfoHash, p bittorrent.Peer)
 	return idx
 }
 
+func (s *peerStore) shardIndexByAddressType(infoHash bittorrent.InfoHash, v6 bool) uint32 {
```

Comment:
Yeah I thought about merging/chaining them, I'll do it.

About the index: We don't do that to avoid collisions - we simply use the first half of all shards for IPv4 swarms and the second half for IPv6 swarms.